features/step_definitions.rb

```diff
@@ -163,7 +163,13 @@
     else
       {}
     end
-  config[key] = YAML.load(value)
+
+  if Gem::Version.new(Psych::VERSION) >= Gem::Version.new("4.0.0")
+    config[key] = YAML.load(value, permitted_classes: [Date])
+  else
+    config[key] = YAML.load(value)
+  end
```

Comment:
As suggested in the last part of the PR description, this could also be:

```suggestion
  config[key] = YAML.unsafe_load(value)
```